File: SeleniumExercise/src/main/java/applicationpages/manager/ManagerPage.java
package applicationpages.manager;

import applicationpages.BasePage;
import java.util.ArrayList;
import java.util.List;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;

public class ManagerPage extends BasePage {
	@FindBy(how = How.XPATH, using = "//button/span[text()='Create']")
	private WebElement createButton;

	@FindBy(how = How.XPATH, using = "//li[text()='Folder']")
	private WebElement folderOption;

	@FindBy(how = How.XPATH, using = "//li[text()='Process']")
	private WebElement processOption;

	@FindBy(how = How.XPATH, using = "//div[@aria-label='Select Folder']/div[1]")
	private List<WebElement> foldersList;

	@FindBy(how = How.XPATH, using = "//div[contains(@class,'ProcessListInnerContainer')]/div/div[contains(@class,'ProcessListRowItem')]")
	private List<WebElement> processesList;

	public ManagerPage(WebDriver driver) {
		super(driver);
	}

	public ManagerPage clickCreateButton() {
		createButton.click();
		return new ManagerPage(driver);
	}

	public CreateModalPage selectFolderOption() {
		folderOption.click();
		return new CreateModalPage(driver);
	}

	public CreateModalPage selectProcessOption() {
		processOption.click();
		return new CreateModalPage(driver);
	}

	public List<String> getFoldersList() {
		List<String> folderNames = new ArrayList<>();
		for (WebElement folder : foldersList) {
			folderNames.add(folder.getText());
		}
		return folderNames;
	}

	public ManagerPage clickFolder(String folderName) {
		for (WebElement folder : foldersList) {
			if (folder.getText().equals(folderName)) {
				folder.click();
				break;
			}
		}
		return new ManagerPage(driver);
	}


	public ProcessPage clickProcess(String processName){
		for (WebElement process : processesList) {
			if (process.getText().equals(processName)) {
				process.click();
				break;
			}
		}
		return new ProcessPage(driver);
	}
}

File: SeleniumExerciseC#/Pages/manager/ManagerPage.cs
using OpenQA.Selenium;
using SeleniumExtras.PageObjects;

namespace Pages
{
    class ManagerPage : BasePage
    {
        [FindsBy(How = How.XPath, Using = "//button/span[text()='Create']")]
        private IWebElement createButton;

        [FindsBy(How = How.XPath, Using = "//li[text()='Folder']")]
        private IWebElement folderOption;

        [FindsBy(How = How.XPath, Using = "//li[text()='Process']")]
        private IWebElement processOption;

        [FindsBy(How = How.XPath, Using = "//div[@aria-label='Select Folder']/div[1]")]
        private IList<IWebElement> foldersList;

        [FindsBy(How = How.XPath, Using = "//div[contains(@class,'ProcessListInnerContainer')]/div/div[contains(@class,'ProcessListRowItem')]")]
        private IList<IWebElement> processesList;

        public ManagerPage(IWebDriver driver) : base(driver) { }

        public ManagerPage clickCreateButton()
        {
            createButton.Click();
            return new ManagerPage(driver);
        }

        public CreateModalPage selectFolderOption()
        {
            folderOption.Click();
            return new CreateModalPage(driver);
        }

        public CreateModalPage selectProcessOption()
        {
            processOption.Click();
            return new CreateModalPage(driver);
        }

        public List<String> getFoldersList()
        {
            List<String> folderNames = new List<string>();
            foreach (IWebElement folder in foldersList)
            {
                folderNames.Add(folder.Text);
            }
            return folderNames;
        }

        public ManagerPage clickFolder(String folderName)
        {
            foreach (IWebElement folder in foldersList)
            {
                if (folder.Text.Equals(folderName))
                {
                    folder.Click();
                    break;
                }
            }
            return new ManagerPage(driver);
        }

        public ProcessPage clickProcess(String processName)
        {
            foreach (IWebElement process in processesList)
            {
                if (process.Text.Equals(processName))
                {
                    process.Click();
                    break;
                }
            }
            return new ProcessPage(driver);
        }
    }
}